dbt/adapters/redshift/connections.py

```diff
@@ -253,9 +254,8 @@ def cancel(self, connection: Connection):
             raise
 
         sql = f"select pg_terminate_backend({pid})"
-        _, cursor = self.add_query(sql)
-        res = cursor.fetchone()
-        logger.debug(f"Cancel query '{connection.name}': {res}")
+        cursor = connection.handle.cursor()
+        cursor.execute(sql)
```

Comment:
This feels like the right way to handle this anyway; if we're cancelling a query, we're probably already in a failed state and don't necessarily need results from the database. However, can we keep the log event in here? Maybe make it something like:

```python
logger.debug(f"Cancel query on: '{connection.name}' with PID: {pid}")
```

Reply:
It looks like going from `add_query` to `execute` causes this to be dropped from the observed call stack in the mock. The unit tests fails because it doesn't see the call from `execute`. From looking at the unit test you updated below, it seems like you might just need to mock `execute` as well (but also in the other tests where it failed).

Reply:
Removed it from the expected call stack in https://github.com/dbt-labs/dbt-redshift/pull/568/commits/a36d7f89dfcd12b23d56f3518cb96ca59a0b0de5, so passing the remaining tests.

But that particular call is no longer asserted within the unit tests now. Do you want to leave it as-is, or would you prefer to try to add a mock for `execute` as well?

Reply:
Oh funny, for some reason I thought it was more than one test. I guess it was just he different os/py versions. Looks good to me then.